src/common/src/catalog/internal_table.rs

```diff
@@ -38,3 +39,35 @@ pub fn valid_table_name(table_name: &str) -> bool {
         LazyLock::new(|| Regex::new(r"__internal_.*_\d+").unwrap());
     !INTERNAL_TABLE_NAME.is_match(table_name)
 }
+
+pub fn get_dist_key_in_pk_indices(dist_key_indices: &[usize], pk_indices: &[usize]) -> Vec<usize> {
+    let dist_key_in_pk_indices = dist_key_indices
+        .iter()
+        .map(|&di| {
+            pk_indices
+                .iter()
+                .position(|&pi| di == pi)
+                .unwrap_or_else(|| {
+                    panic!(
+                        "distribution key {:?} must be a subset of primary key {:?}",
+                        dist_key_indices, pk_indices
+                    )
+                })
+        })
+        .collect_vec();
+    dist_key_in_pk_indices
+}
+
+/// Get distribution key start index in pk, and return None if `dist_key_in_pk_indices` is not empty
+/// or continuous.
+/// Note that `dist_key_in_pk_indices` may be shuffled, the start index should be the
+/// minimum value.
+pub fn get_dist_key_start_index_in_pk(dist_key_in_pk_indices: &[usize]) -> Option<usize> {
```

Comment:
`get_min_dist_key_start_index_if_continuous`?

Reply:
start_index is always min